dll/win32/browseui/addresseditbox.cpp

```diff
@@ -94,6 +95,72 @@ HRESULT STDMETHODCALLTYPE CAddressEditBox::SetCurrentDir(long paramC)
     return E_NOTIMPL;
 }
 
+BOOL CAddressEditBox::GetComboBoxText(CComHeapPtr<WCHAR>& pszText)
+{
+    pszText.Free();
+    INT cchMax = fCombobox.GetWindowTextLength() + sizeof(UNICODE_NULL);
+    if (!pszText.Allocate(cchMax))
+        return FALSE;
+    return fCombobox.GetWindowText(pszText, cchMax);
+}
+
+HRESULT CAddressEditBox::GetAbsolutePidl(PIDLIST_ABSOLUTE *pAbsolutePIDL)
+{
+    CComPtr<IBrowserService> isb;
+    HRESULT hr = IUnknown_QueryService(fSite, SID_STopLevelBrowser, IID_PPV_ARG(IBrowserService, &isb));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    hr = isb->GetPidl(pAbsolutePIDL);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    return S_OK;
+}
+
+/* Execute command line from address bar */
+BOOL CAddressEditBox::ExecuteCommandLine()
+{
+    /* Get command line */
+    CComHeapPtr<WCHAR> pszCmdLine;
+    if (!GetComboBoxText(pszCmdLine))
+        return FALSE;
+
+    /* Split 1st parameter from trailing arguments */
+    PWCHAR args = PathGetArgsW(pszCmdLine);
+    if (args && *args)
+    {
+        --args;
+        *args = UNICODE_NULL;
+        ++args;
+    }
+
+    PathUnquoteSpacesW(pszCmdLine); /* Unquote the 1st parameter */
+
+    /* Get ready for execution */
+    SHELLEXECUTEINFOW info = { sizeof(info), SEE_MASK_FLAG_NO_UI, m_hWnd };
+    info.lpFile = pszCmdLine;
+    info.lpParameters = args;
+    info.nShow = SW_SHOWNORMAL;
+
+    WCHAR dir[MAX_PATH] = L"";
+    CComHeapPtr<ITEMIDLIST> pidl;
+    if (SUCCEEDED(GetAbsolutePidl(&pidl)))
+    {
+        if (SHGetPathFromIDListW(pidl, dir) && PathIsDirectoryW(dir))
+            info.lpDirectory = dir; /* Set current directory */
```

Comment:
Maybe you should instead move that comment at the beginning of that whole block of code (WCHAR dir... etc) : this will allow to explain why you are doing this whole work of getting a pidl etc.

Reply:
Moved.